flink-cdc-common/src/main/java/org/apache/flink/cdc/common/data/binary/BinaryArrayData.java

```diff
@@ -0,0 +1,623 @@
+    private static final int DOUBLE_ARRAY_OFFSET = UNSAFE.arrayBaseOffset(double[].class);
+
+    /**
+     * Calculates the size of the header in bytes for an array with the specified number of fields.
+     *
+     * <p>The header consists of:
+     *
+     * <ul>
+     *   <li>4 bytes to store the size of the array (number of elements).
+     *   <li>A bitmap to track null values, where each bit represents whether an element is null.
+     *       This bitmap is aligned to 4-byte word boundaries for efficient memory access and to
+     *       facilitate the use of bitwise operations.
+     * </ul>
+     *
+     * <p>The size of the bitmap is determined by the number of elements in the array:
+     *
+     * <ul>
+     *   <li>Each element requires 1 bit in the bitmap.
+     *   <li>The total number of bits is rounded up to the nearest multiple of 32 to ensure
+     *       alignment to 4-byte word boundaries (i.e., a 32-bit integer).
+     * </ul>
+     *
+     * <p>The formula for calculating the size of the header is:
+     *
+     * <pre>
+     *   header size = 4 bytes (for array size) + ((numFields + 31) / 32) * 4 bytes (for null bitmap)
+     * </pre>
+     *
+     * @param numFields the number of elements in the array
+     * @return the size of the header in bytes
+     */
+    public static int calculateHeaderInBytes(int numFields) {
+        return 4 + ((numFields + 31) / 32) * 4;
+    }
+
+    /**
+     * It store real value when type is primitive. It store the length and offset of variable-length
+     * part when type is string, map, etc.
+     */
+    public static int calculateFixLengthPartSize(DataType type) {
+        // ordered by type root definition
+        switch (type.getTypeRoot()) {
+            case BOOLEAN:
+            case TINYINT:
+                return 1;
+            case CHAR:
+            case VARCHAR:
+            case BINARY:
+            case VARBINARY:
+            case DECIMAL:
+            case BIGINT:
+            case DOUBLE:
+            case TIMESTAMP_WITHOUT_TIME_ZONE:
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+            case ARRAY:
+            case MAP:
+            case ROW:
+                // long and double are 8 bytes;
+                // otherwise it stores the length and offset of the variable-length part for types
+                // such as is string, map, etc.
+                return 8;
+            case TIMESTAMP_WITH_TIME_ZONE:
+                throw new UnsupportedOperationException();
+            case SMALLINT:
+                return 2;
+            case INTEGER:
+            case FLOAT:
+            case DATE:
+            case TIME_WITHOUT_TIME_ZONE:
+                return 4;
+            default:
+                throw new IllegalArgumentException();
+        }
+    }
+
+    // The number of elements in this array
+    private int size;
+
+    /** The position to start storing array elements. */
```

Comment:
Minor comment, would you like to unify code comment format? Seems here use two comment formats.